int32 numberOfLines
LineSegment[] lineSegmentArray